src/api/app/models/linked_project.rb

```diff
@@ -2,20 +2,52 @@ class LinkedProject < ApplicationRecord
   belongs_to :project, foreign_key: :db_project_id
   belongs_to :linked_db_project, class_name: 'Project', optional: true
 
-  validate :validate_duplicates
+  validates :linked_db_project, presence: true, unless: -> { linked_remote_project_name.present? }
+  validates :linked_remote_project_name, presence: true, unless: -> { linked_db_project.present? }
+  validates :linked_remote_project_name, length: { maximum: 255 }
+  validates :db_project_id, uniqueness: {
+    scope: :linked_db_project_id,
+    if: -> { linked_db_project_id.present? },
+    message: ->(object, _data) { "already linked with '#{object.linked_db_project}'" },
+    on: :create
+  }
+  validates :db_project_id, uniqueness: {
+    scope: :linked_remote_project_name,
+    if: -> { linked_remote_project_name.present? },
+    message: ->(object, _data) { "already linked with '#{object.linked_remote_project_name}'" },
+    on: :create
+  }
+  validate :validate_target
+  validate :validate_cycles
+  validate :validate_access_flag_equality
 
   protected
 
-  def validate_duplicates
-    if !project
-      errors.add(:project, 'Can not link project to not existing project')
-    elsif !linked_db_project && !linked_remote_project_name
-      errors.add(:linked_db_project, 'It must be linked to somewhere')
-    elsif linked_db_project && linked_remote_project_name
-      errors.add(:linked_remote_project_name, "Can't have a local and a remote link")
-    elsif LinkedProject.find_by(project: project, linked_db_project: linked_db_project)
-      errors.add(:project, 'Already linked with that project')
-    end
+  def validate_target
+    return unless linked_db_project && linked_remote_project_name
+
+    errors.add(:base, 'can not have both linked_db_project and linked_remote_project_name')
+  end
+
+  def validate_cycles
```

Comment:
```suggestion
  def validate_circular_dependency
```
I think that's the [standard name](https://en.wikipedia.org/wiki/Circular_dependency)

Reply:
not sure what you mean...

Reply:
It was about the `cycles` it took me a while to understand the code was ensuring no circular dependencies were set, so I thought that changing the `validate_cycles` to `validate_circular_dependency` was a better name.